numpy/core/src/umath/ufunc_object.c

```diff
@@ -2089,6 +2107,124 @@ PyUFunc_GeneralizedFunction(PyUFuncObject *ufunc,
         goto fail;
     }
 
+    if (axis) {
+        /*
+         * possibly remap axes.
+         */
+        int axis_rqst;
```

Comment:
Mostly since your `check_and_adjust_axis` seems to want an `int`. I'm eternally confused about these C types...